core/build.gradle

```diff
@@ -33,31 +32,16 @@ shadowJar {
         'META-INF/services/com.fasterxml.jackson.core.*',
         'META-INF/services/com.github.dockerjava.api.command.*',
         'META-INF/services/javax.ws.rs.ext.*',
-        'META-INF/services/org.glassfish.hk2.extension.*',
-        'META-INF/services/org.jvnet.hk2.external.generator.*',
-        'META-INF/services/org.glassfish.jersey.internal.spi.*',
         'META-INF/services/java.security.Provider',
         'mozilla/public-suffix-list.txt',
     ].each { exclude(it) }
 
-    dependencies {
-        include(dependency('org.apache.httpcomponents:.*'))
-        include(dependency('org.glassfish.*:.*'))
-        include(dependency('org.aopalliance.*:.*'))
-        include(dependency('javax.ws.rs:.*'))
-        include(dependency('com.fasterxml.*:.*'))
-        include(dependency('com.github.docker-java:.*'))
-        include(dependency('com.google.guava:.*'))
-        include(dependency('io.netty:.*'))
-        include(dependency('org.bouncycastle:.*'))
-        include(dependency('org.zeroturnaround:zt-exec'))
-        include(dependency('commons-lang:commons-lang'))
-        include(dependency('commons-io:commons-io'))
-        include(dependency('commons-codec:commons-codec'))
-        include(dependency('com.squareup.okhttp3:.*'))
-        include(dependency('com.squareup.okio:.*'))
-        include(dependency('org.scala-sbt.ipcsocket:ipcsocket'))
-        include(dependency('org.glassfish.main.external:trilead-ssh2-repackaged'))
+    project.afterEvaluate {
+        dependencies {
+            for (id in project.configurations.compile.resolvedConfiguration.resolvedArtifacts*.moduleVersion*.id) {
+                exclude(dependency("${id.group}:${id.name}"))
```

Comment:
This is interesting - what's this doing? It looks like we're excluding all transitive deps. Is this in order to make all deps explicit?

Reply:
Read it as: "shade everything, except the dependencies from the `compile` scope". It means that if docker-java adds a dependency on X, it will never appear in our dependencies unless we add it to `compile` scope (aka add it as a public dependency of the library)

Reply:
Got it 👍